### DISCUSSION TITLE
Does running "npm run build" verify routes load?

### DISCUSSION THREAD

**Discussion by pkellner** [vercel/next.js#13115] [Help] [answered]
I've noticed that I have a runtime JavaScript error on any route in my app, the "npm run build" fails with an error.

What intelligent statement can I make if "npm run build" runs without an error?

**Comment by timneutkens** [comment, marked as answer]
> I've noticed that I have a runtime JavaScript error on any route in my app, the "npm run build" fails with an error.

You might run into this if the pages use static generation as that executes the code during `next build`. In order to verify if a page is static we require the page and read it's exports.

**Reply by pkellner** [reply]
that makes sense. Hope you are well Tim. Enjoyed your webcast the other week.

**Reply by timneutkens** [reply]
I am! Hope you're well too! 🙏
